From ef4523354f8ecb87cb83b992ac8dd92d0391f1e8 Mon Sep 17 00:00:00 2001
From: Markus Bauer <mb@karo-electronics.de>
Date: Wed, 4 Nov 2020 14:36:09 +0100
Subject: [PATCH] net: stmmac: Don't log error message in case of -EPROBE_DEFER.

Remove error messages that might confuse users when error is just -517 / -EPROBE_DEFER.

[...]
imx-dwmac 30bf0000.ethernet: Cannot register the MDIO bus                                                                          
imx-dwmac 30bf0000.ethernet: stmmac_dvr_probe: MDIO bus (id: 0) registration failed
[...]

---
 drivers/net/ethernet/stmicro/stmmac/stmmac_main.c | 7 ++++---
 drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c | 6 ++++--
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
index 33272a12989a..7d1cdd576b91 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c
@@ -4857,9 +4857,10 @@ int stmmac_dvr_probe(struct device *device,
 		/* MDIO bus Registration */
 		ret = stmmac_mdio_register(ndev);
 		if (ret < 0) {
-			dev_err(priv->device,
-				"%s: MDIO bus (id: %d) registration failed",
-				__func__, priv->plat->bus_id);
+			if (ret != -EPROBE_DEFER)
+				dev_err(priv->device,
+					"%s: MDIO bus (id: %d) registration failed, err=%d",
+					__func__, priv->plat->bus_id, ret);
 			goto error_mdio_register;
 		}
 	}
diff --git a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
index 226e5a4bf21c..8e202f63da31 100644
--- a/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
+++ b/drivers/net/ethernet/stmicro/stmmac/stmmac_mdio.c
@@ -401,8 +401,10 @@ int stmmac_mdio_register(struct net_device *ndev)
 	new_bus->parent = priv->device;
 
 	err = of_mdiobus_register(new_bus, mdio_node);
-	if (err != 0) {
-		dev_err(dev, "Cannot register the MDIO bus\n");
+	if (err) {
+		if (err != -EPROBE_DEFER)
+			dev_err(dev,
+				"Cannot register the MDIO bus, err=%d\n", err);
 		goto bus_register_fail;
 	}
 
-- 
2.17.1

